metaflow/metaflow_git.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+"""Get git repository information for the package
+
+Functions to retrieve git repository details like URL, branch name, 
+and commit SHA for Metaflow code provenance tracking.
+"""
+
+import os
+import subprocess
+from os import path, name, environ
+from typing import Dict, Union
+
+# Cache for git information to avoid repeated subprocess calls
+_git_info_cache = None
+
+__all__ = ("get_repository_info",)
+
+GIT_COMMAND = "git"
+
+
+def _get_repo_url(path: Union[str, os.PathLike]):
+    """Get the repository URL from git config"""
+    try:
+        result = subprocess.run(
+            [GIT_COMMAND, "config", "--get", "remote.origin.url"],
+            cwd=path,
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        if result.returncode == 0:
+            url = result.stdout.strip()
+            # Convert SSH URLs to HTTPS for clickable links
+            if url.startswith("git@"):
+                parts = url.split(":")
```

Comment:
Is there any case where we should consider: `url.split(":", 1)`? Very rare edge case.

Reply:
good catch, will add.